rotkehlchen/chain/ethereum/modules/ens/decoder.py

```diff
@@ -139,17 +140,25 @@ def _decode_name_renewed(self, context: DecoderContext) -> DecodingOutput:
             return DEFAULT_DECODING_OUTPUT
 
         name = decoded_data[0]
-        amount = from_wei(decoded_data[1])
+        name_cost = from_wei(decoded_data[1])
         expires = decoded_data[2]
 
-        for event in context.decoded_events:
+        refund_event_idx, refund_amount = None, ZERO
+        for idx, event in enumerate(context.decoded_events):
+            if event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.NONE and event.asset == A_ETH and event.address == context.tx_log.address:  # noqa: E501
+                refund_event_idx = idx
+                refund_amount = event.balance.amount
+
             # Find the transfer event which should be before the name renewed event
-            if event.event_type == HistoryEventType.SPEND and event.asset == A_ETH and event.balance.amount == amount and event.address == context.tx_log.address:  # noqa: E501
+            if event.event_type == HistoryEventType.SPEND and event.asset == A_ETH and event.balance.amount - refund_amount == name_cost and event.address == context.tx_log.address:  # noqa: E501
+                event.balance.amount -= refund_amount
                 event.event_type = HistoryEventType.RENEW
                 event.event_subtype = HistoryEventSubType.NFT
                 event.counterparty = CPT_ENS
-                event.notes = f'Renew ENS name {name} for {amount} ETH until {self.timestamp_to_date(expires)}'  # noqa: E501
+                event.notes = f'Renew ENS name {name} for {name_cost} ETH until {self.timestamp_to_date(expires)}'  # noqa: E501
```

Comment:
Shouldn't this be also reflected?
```suggestion
                event.notes = f'Renew ENS name {name} for {event.balance.amount} ETH until {self.timestamp_to_date(expires)}'  # noqa: E501
```

Reply:
As said in the other comment, `name_cost` = `full amount spent` - `refund amount`.

`event.balance.amount` is first equal to `full amount spent`, but as you can see a few lines above, `refund amount` is subtracted from it. So here `event.balance.amount` is equal to `name_cost`, which is the actual price of the ens name.

Reply:
This is confusing. Should add a small comment above : `event.balance.amount -= refund_amount  # is now equal to name_cost`